tripal_chado/src/Plugin/Field/FieldType/ChadoOrganismTypeDefault.php

```diff
@@ -3,131 +3,259 @@
-      //'chado_column' => $base_fk_col,
+
+    // Base table links directly
+    if ($base_table == $linker_table) {
+      $properties[] = new ChadoIntStoragePropertyType($entity_type_id, self::$id, self::$object_id, $linker_fkey_term, [
+        'action' => 'store',
+        'drupal_store' => TRUE,
+        'path' => $base_table . '.' . $linker_fkey_column,
+        'delete_if_empty' => TRUE,
+        'empty_value' => 0,
+      ]);
+    }
+    // An intermediate linker table is used
+    else {
+      // Define the linker table that links the base table to the object table.
+      $properties[] = new ChadoIntStoragePropertyType($entity_type_id, self::$id, 'linker_id', $record_id_term, [
+        'action' => 'store_pkey',
+        'drupal_store' => TRUE,
+        'path' => $linker_table . '.' . $linker_pkey_col,
+      ]);
+
+      // Define the link between the base table and the linker table.
+      $properties[] = new ChadoIntStoragePropertyType($entity_type_id, self::$id, 'link', $linker_left_term, [
+        'action' => 'store_link',
+        'drupal_store' => FALSE,
+        'path' => $base_table . '.' . $base_pkey_col . '>' . $linker_table . '.' . $linker_left_col,
+      ]);
+
+      // Define the link between the linker table and the object table.
+      $properties[] = new ChadoIntStoragePropertyType($entity_type_id, self::$id, self::$object_id, $linker_fkey_term, [
+        'action' => 'store',
+        'drupal_store' => TRUE,
+        'path' => $linker_table . '.' . $linker_fkey_column,
+        'delete_if_empty' => TRUE,
+        'empty_value' => 0,
+      ]);
+
+      // Other columns in the linker table. Set in the widget, but currently not implemented in the formatter.
+      // Typically these are type_id and rank, but are not present in all linker tables,
+      // so they are added only if present in the linker table.
+      foreach ($extra_linker_columns as $column => $term) {
+        $properties[] = new ChadoIntStoragePropertyType($entity_type_id, self::$id, 'linker_' . $column, $term, [
+          'action' => 'store',
+          'drupal_store' => FALSE,
+          'path' => $linker_table . '.' . $column,
+          'as' => 'linker_' . $column,
+        ]);
+      }
+    }
+
+    // The object table, the destination table of the linker table
+    $properties[] = new ChadoVarCharStoragePropertyType($entity_type_id, self::$id, 'organism_genus', $genus_term, $genus_len, [
+      'action' => 'read_value',
+      'drupal_store' => FALSE,
+      'path' => $linker_table . '.' . $linker_fkey_column . '>' . $object_table . '.' . $object_pkey_col . ';genus',
+      'chado_table' => $object_table,
+      'chado_column' => 'genus',
+      'as' => 'organism_genus',
     ]);
-    $properties[] =  new ChadoVarCharStoragePropertyType($entity_type_id, self::$id, 'label', $label_term, $label_len, [
-      'action' => 'replace',
-      'template' => "<i>[genus] [species]</i> [infraspecific_type] [infraspecific_name]",
```

Comment:
I'm just marking this as a topic of dicussion for the Codefest day 2.

Reply:
So, @laceysanderson and I had a conversation abou this as the codefest was wrapping up.  We think this property should come back but have a key of 'scientific_name' instead.

Reply:
This is done with commit https://github.com/tripal/tripal/pull/1743/commits/720af13025158a14f6edc9d1a9f45d8458bcdaea